roku/device/roku-tv.groovy

```diff
@@ -321,21 +324,25 @@ void scheduleQueryActiveApp() {
 }
 
 void scheduleQueryMediaPlayer() {
-    unschedule('queryMediaPlayer')
+    if (this[SETTING_LOG_ENABLE]) { log.info "scheduleQueryMediaPlayer: Executed " }
+//    unschedule('queryMediaPlayer')
 
     Long delay = (this[SETTING_MEDIA_INTERVAL] ?: 0) * (this[SETTING_MEDIA_UNITS] == REFRESH_UNIT_MINUTES ? 60 : 1)
 
-    if (device.currentValue('application', true) == 'Roku') {
+    if (device.latestValue('application', true) == 'Roku') {
```

Comment:
device.currentValue('application'), device.latestValue('application') and device.currentApplication all return the same value, and behave the same.  As such, the change from current to latest does not impact the behavior.  The use of the skipCache might.  I need to see if I have any functions or variables that conflict with this naming convention.

Reply:
I included this based on our previous conversations when trying to diagnose why the Media Player scheduler wasn't working. With that said i am not sure if it made a difference or not, I tend to believe that for some reason the addition of the runin call to execute scheduleQueryMediaPlayer on line 642 is more likely what resolved this.

Reply:
yes.  I am just letting you know that I just learned they are all identical calls -- something ported over from SmartThings that was phased out on ST as well, but is here for portability.  I was mislead from a forum post that they were different, but now I know they are the same.  the HE preferred command is currentValue or current_Attribute_.  I am sure the only thing that made a different is that delayed call.  I may need to rethink how this works.  On an older version of the Roku OS, when I switched from App1 to App2, it would not go from App1 to Roku to App2 -- maybe a timing thing in the old driver.  But now it does, and I think different TVs have different performance resulting in your behavior being different than mine.  So, I think I need a better way of managing the schedulers so that I can add/remove/add back-to-back w/o issues, regardless of the timing.  I think I have a solution.